### PR TITLE
feature(serializer): support encoding for number primitives under Compact<T>

### PR DESCRIPTION
**Pull request by pandres95**

### CODE REVIEW

File: src/serializer.rs [hunk 1]
```diff
@@ -133,6 +134,34 @@ where
...
+        let type_def = self
+            .registry
+            .expect("registry not present")
+            .resolve(ty)
+            .expect("type T for Compact<T> not found in registry")
+            .type_def();
+
+        let compact_buffer = if let scale_info::TypeDef::Primitive(p) = type_def {
+            use codec::Compact;
+            if matches!(p, TypeDefPrimitive::U32) {
```
Comment (olanod) [inline src/serializer.rs]: We can replace the whole block with a single `match` block. We could even prevent the the nested `if` doing `Primitive(TypeDefPrimitive::X)`

File: src/serializer.rs [hunk 2]
```diff
@@ -133,6 +134,34 @@ where
         };
         Serializer { out, ty, registry }
     }
+
+    fn serialize_compact(&mut self, ty: u32, v: u128) -> Result<()> {
+        let type_def = self
+            .registry
+            .expect("registry not present")
```
Comment (olanod) [inline src/serializer.rs]: `expect`(and `unwrap`) should be avoided when possible specially when we are already returning a `Result`.

**Review by olanod — CHANGES_REQUESTED** [review]

File: src/serializer.rs (hunk 1 above)
Comment (pandres95) [reply src/serializer.rs]: The problem with using a `match` block, is that it'll require to fulfill all matching arms behind `TypeDef`. That includes things that will never be children of a `Compact`, like `Struct`s, `Composite`s, `Variant`s, and even other `Compact`s. 

I could, however, prevent the nested if, by doing `Primitive(TypeDefPrimitive::UXX)`. Thanks for that suggestion.

File: src/serializer.rs (hunk 2 above)
Comment (pandres95) [reply src/serializer.rs]: Ended up using `self.resolve`, as it did the same thing as this snippet.

Paradoxically, this function also uses `expect` 🤭.

**Review by olanod — CHANGES_REQUESTED** [review]
I made the change I suggested before of using match instead of matches :)
Something that would be left is replacing the codec dependency to correctly compile with `--no-default-features` and some tests of course ;)

**Review by olanod — APPROVED** [review]

**Comment by olanod** [comment]
@pandres95 I created https://github.com/virto-network/scales/issues/8 to track removing the dependency on scale-codec

**Pull request merged**